### ISSUE TITLE
Any plan to support C codebase?

### ISSUE THREAD

**Issue by rafsun42** [gabotechs/dep-tree#75]
We can already generate doxygen to generate dependency graph. But, 3d interactive graph looks cool.

**Comment by Attempt3035** [comment]
Agreed here! C and C++ would be awesome, even if it's an adaptor built on existing dep discovery systems that just plugs into the 3D force graph!

**Comment by gabotechs** [comment]
There's currently no plans for implementing C or C++ in the short term, but it looks like it's the most requested language.

I would first try to document how to implement new languages rather than implement them myself, to see if that attracts some contributions, as right now I cannot afford to deviate too much from the functionalities that I use in my day-to-day work, but if/when I have some buffer I will definitely try.

**Comment by gabotechs** [comment]
I have created a guide about how to contribute new languages to Dep Tree https://github.com/gabotechs/dep-tree/blob/main/docs/IMPLEMENTING_NEW_LANGUAGES.md.

If anyone wants to have some fun with it, be my guest.
